tools/clang/test/HLSLFileCheck/hlsl/signature/sig-flatten-struct-array.hlsl

```diff
@@ -0,0 +1,52 @@
+// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s
+
+// Check correct assignment of semantic indices to parameter with array of
+// struct followed by another argument, which was triggering an assert
+// and invalid indexing for updating the index in the annotation.
+
+// CHECK: ; Output signature:
+// CHECK-NEXT: ;
+// CHECK-NEXT: ; Name                 Index   Mask Register SysValue  Format   Used
+// CHECK-NEXT: ; -------------------- ----- ------ -------- -------- ------- ------
+// CHECK-NEXT: ; OUT                      0   x           0     NONE   float   x
+// CHECK-NEXT: ; OUT                      1    y          0     NONE   float    y
+// CHECK-NEXT: ; OUT                      3   x           1     NONE   float   x
+// CHECK-NEXT: ; OUT                      2    y          1     NONE   float    y
+// CHECK-NEXT: ; OUT                      6   x           2     NONE   float   x
+// CHECK-NEXT: ; OUT                      4    y          2     NONE   float    y
+// CHECK-NEXT: ; OUT                      5    y          3     NONE   float    y
+// CHECK-NEXT: ; OUT                      7    y          4     NONE   float    y
+// CHECK-NEXT: ; OUT                      8    y          5     NONE   float    y
+// CHECK-NEXT: ; OUT                      9   xyzw        6     NONE   float   xyzw
```

Comment:
Would be nice to add comment about each line like
// Out.foo[0].a
// CHECK-NEXT: ; OUT                      0   x           0     NONE   float   x
// Out.foo[0].b[0]
// CHECK-NEXT: ; OUT                      1    y          0     NONE   float    y
...
// Out.postion
// CHECK-NEXT: ; OUT                      9   xyzw        6     NONE   float   xyzw

Reply:
done.